### PR TITLE
Use popup instead of DOM insertion

### PR DESCRIPTION
**Pull request by pamelafox**
I attempted fixing #2 again but the GitHub UI is now rendering multiple copies of elements and worse, I couldn't get the z-index higher than their elements. So I propose using popups from the browser icon - more reliable and maintable. This PR makes that change.

### CODE REVIEW

File: src/index.jsx [hunk 1]
```diff
@@ -13,10 +13,16 @@ import { AVAILABLE_APPS, generateRegularUrl } from './generator';
 import { getPref, setPref } from './prefs';
 
 function copyGeneratedUrl(hubUrl, app) {
-    const parts = GitUrlParse(window.location.href);
-    const repoUrl = `${parts.protocol}://${parts.source}/${parts.full_name}`;
-    const url = generateRegularUrl(hubUrl, repoUrl, parts.ref, app, parts.name + '/' + parts.filepath);
-    navigator.clipboard.writeText(url);
+    var query = { active: true, currentWindow: true };
```
Comment (yuvipanda) [inline src/index.jsx]:
```suggestion
    const query = { active: true, currentWindow: true };
```

File: src/index.jsx [hunk 2]
```diff
@@ -13,10 +13,16 @@ import { AVAILABLE_APPS, generateRegularUrl } from './generator';
...
+    chrome.tabs.query(query, function (tabs) {
+        var activeTab = tabs[0];
```
Comment (yuvipanda) [inline src/index.jsx]:
```suggestion
        const activeTab = tabs[0];
```

**Review by yuvipanda — COMMENTED** [review]
Thanks for continuing to work on this, @pamelafox!

I agree this seems generally much more maintainable than whack-a-mole-ing with GitHub's structure.

Can you update the screenshots and help text in the README to match? I think the UI would put the button in a less prominent place too, right?

**Comment by pamelafox** [comment]
I've updated the readme, but also I've given it a custom icon based on the logo file that was in the repo, and I disable it on non-GitHub URLs.

GitHub:
Screenshot 2023-08-20 at 7 27 24 AM

Not GitHub:
Screenshot 2023-08-20 at 7 27 30 AM

File: src/background.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+chrome.tabs.onUpdated.addListener((tabId, changeInfo, tab) => {
+    console.log(tab.url);
```
Comment (yuvipanda) [inline src/background.js]:
```suggestion
```

**Review by yuvipanda — APPROVED** [review]
This looks pretty great and much more long term maintainable! It does lead to the end of perhaps the best function name I have ever written, but I think the world is a better place for that :D Thank you for cleaning all this up :)

The last question I have before merging (other than one small code suggestion) - have you tested this on both Firefox and Chrome? If so, I'll merge and make a release!

Thank you so much!

**Comment by yuvipanda** [comment]
Also /cc @balajialg and @ericvd-ucb

**Comment by pamelafox** [comment]
@yuvipanda Good call on testing in FF! I discovered that FF and Chrome cant agree on background scripts, so you have to have two different maifest files: https://github.com/mozilla/web-ext/issues/2532
After much webpack shenanigans, I now have it outputting to 2 different dist folders and you can point build at that.
Yay web standards or lack thereof!

**Comment by yuvipanda** [comment]
Unfortunately on firefox I can't get the extension to load still :(

image

**Comment by yuvipanda** [comment]
And if I load it on Chrome, it loads but there's no icon. Chrome tells me that perhaps the extension is not active on this page? I was on https://github.com/yuvipanda/example-inherit-from-community-image

image

**Comment by pamelafox** [comment]
Hm, I'm not sure why FF says its corrupt. I've changed the readme to use a different output dir for firefox vs. chrome in case that was the issue. 
I've recorded a screencast of me loading the FF one locally:
https://youtu.be/0kjyyB9C77o
And then the Chrome one:
https://youtu.be/g8tV5DA83Tk

To get the icon on the toolbar, you need to click that pushpin to pin it. That's how it works in Chrome these days, it appears: https://stackoverflow.com/questions/64028193/pin-chrome-extension-icon-by-default-programmatically

**Comment by balajialg** [comment]
@pamelafox @yuvipanda Any updates on this PR? Is it possible to push the changes to the repo even if it works only for the Chrome browser? Few instructors at Berkeley would love to test the latest changes to this extension. Thanks!

**Comment by yuvipanda** [comment]
Given I'm swamped, I've just invited both @balajialg and @pamelafox to this repo. If you create an account on addons.mozilla.org, I can give you access to publishing the extension there as well. Same for chrome - if you can give me an email address of a google account you'd like to be associated with this, I'll get that account access to push to the chrome store. Hope that works?

**Comment by balajialg** [comment]
Thanks Yuvi, Can you please give access to this email id -
balajialg@gmail.com Thanks!

**Comment by yuvipanda** [comment]
Looks like something ate that email, @balajialg

**Comment by balajialg** [comment]
Ah, thanks Yuvi! I fixed it (I tried responding to this issue via email earlier which removed the email id)

**Comment by pamelafox** [comment]
I log into both with pamela.fox @ gmail , thanks!

**Comment by yuvipanda** [comment]
ok, so I submitted this to addons.mozilla.org even though I can't get it working locally. I've invited @pamelafox on there, but it looks like @balajialg doesn't have an account. Can you create an account there and then I can invite you?

Looks like the google chrome process is more complicated :(

**Comment by yuvipanda** [comment]
I have submitted the chrome extension as well.

**Comment by pamelafox** [comment]
I just tried the latest from both stores and they seem to work for me!
Firefox: https://addons.mozilla.org/en-US/firefox/addon/nbgitpuller-link-generator/
Chrome: https://chromewebstore.google.com/detail/nbgitpuller-link-generato/hpdbdpklpmppnoibabdkkhnfhkkehgnc

I went ahead and updated screenshots on Firefox page to match current user experience. I didn't have access to edit Chrome page.

**Pull request merged**